workspaces/ui-v2/src/optic-components/pages/diffs/CapturePage.tsx

```diff
@@ -129,6 +129,10 @@ export function CapturePage(props: { showDiff?: boolean }) {
   );
 }
 
+export const CapturePageWithDiff = (props: any) => (
```

Comment:
This works for now to reduce rerenders - but I think there's some amount of work we can do with [NavigationRoute](https://github.com/opticdev/optic/blob/96010d79fc65e0b4d5af65a4feb35017384e9e2f/workspaces/ui-v2/src/optic-components/navigation/NavigationRoute.tsx#L14-L31) so we aren't creating new react elements (instead invoking these as functions). This is fine for now though

Reply:
Maybe something like `const MemoedComponent = React.useMemo(() => Component, [path]);` in `NavigationRoute`, but that might not be the best solution.

Reply:
I think the most straightforward thing would be to change `<Component />` to `Component()` - i.e. fn call instead of react.createElement - although there's probably better ways to compose / design these components so we don't need to inject components.